packages/backend/server/src/plugins/copilot/providers/types.ts

```diff
@@ -1,21 +1,21 @@
 import { AiPromptRole } from '@prisma/client';
 import { z } from 'zod';
 
-import { type CopilotProvider } from './provider';
-
 export enum CopilotProviderType {
   FAL = 'fal',
   Gemini = 'gemini',
   OpenAI = 'openai',
   Perplexity = 'perplexity',
 }
 
+export const CopilotProviderSchema = z.object({
+  type: z.nativeEnum(CopilotProviderType),
+});
+
 export enum CopilotCapability {
-  TextToText = 'text-to-text',
-  TextToEmbedding = 'text-to-embedding',
-  TextToImage = 'text-to-image',
-  ImageToImage = 'image-to-image',
-  ImageToText = 'image-to-text',
+  Text = 'text',
+  Embedding = 'embedding',
+  Image = 'image',
```

Comment:
These input types can be combined and arranged. The user's context is not necessarily a single type, but a mixture of various types.